src/Mailgun/Messages/MessageBuilder.php

```diff
@@ -248,7 +248,16 @@ public function addCustomHeader($headerName, $headerData)
         if (!preg_match('/^h:/i', $headerName)) {
             $headerName = 'h:'.$headerName;
         }
-        $this->message[$headerName] = [$headerData];
+
+        if (array_key_exists($headerName, $this->message)) {
+            if (is_array($this->message[$headerName])) {
+                $this->message[$headerName][] = $headerData;
+            } else {
+                $this->message[$headerName] = [$this->message[$headerName], $headerData];
+            }
+        } else {
+            $this->message[$headerName] = $headerData;
+        }
```

Comment:
We could simplify this code bock by always using an array for the headers, even though it is just a single value.  

Wouldn't it also work if this code block was replaced with: 
`$this->message[$headerName][] = $headerData;`